config/main.py

```diff
@@ -7381,5 +7381,126 @@ def date(date, time):
     clicommon.run_command(['timedatectl', 'set-time', date_time])
 
 
+#
+# 'asic-sdk-health-event' group ('config asic-sdk-health-event ...')
+#
+@config.group()
+def asic_sdk_health_event():
+    """Configuring asic-sdk-health-event"""
+    pass
+
+
+@asic_sdk_health_event.group()
+def suppress():
+    """Suppress ASIC/SDK health event"""
+    pass
+
+
+def handle_asic_sdk_health_suppress(db, severity, category_list, max_events, namespace):
+    ctx = click.get_current_context()
+
+    if multi_asic.get_num_asics() > 1:
+        namespace_list = multi_asic.get_namespaces_from_linux()
+    else:
+        namespace_list = [DEFAULT_NAMESPACE]
+
+    severityCapabilities = {
+        "fatal": "REG_FATAL_ASIC_SDK_HEALTH_CATEGORY",
+        "warning": "REG_WARNING_ASIC_SDK_HEALTH_CATEGORY",
+        "notice": "REG_NOTICE_ASIC_SDK_HEALTH_CATEGORY"
+    }
+
+    if category_list:
+        categories = {"software", "firmware", "cpu_hw", "asic_hw"}
+
+        if category_list == 'none':
+            suppressedCategoriesList = []
+        elif category_list == 'all':
+            suppressedCategoriesList = list(categories)
+        else:
+            suppressedCategoriesList = category_list.split(',')
+
+        unsupportCategories = set(suppressedCategoriesList) - categories
+        if unsupportCategories:
+            ctx.fail("Invalid category(ies): {}".format(unsupportCategories))
+
+    if max_events:
+        try:
+            max_events_number = int(max_events)
+            if max_events_number < 0:
+                ctx.fail("Invalid max-events: {}".format(max_events))
+        except ValueError as e:
+            ctx.fail("Invalid max-events: {}".format(max_events))
+
+    for ns in namespace_list:
+        if namespace and namespace != ns:
+            continue
+
+        config_db = db.cfgdb_clients[ns]
+        state_db = db.db_clients[ns]
+
+        entry_name = "SWITCH_CAPABILITY|switch"
+        if "true" != state_db.get(state_db.STATE_DB, entry_name, "ASIC_SDK_HEALTH_EVENT"):
+            ctx.fail("ASIC/SDK health event is not supported on the platform")
+
+        if "true" != state_db.get(state_db.STATE_DB, entry_name, severityCapabilities[severity]):
+            ctx.fail("Suppressing ASIC/SDK health {} event is not supported on the platform".format(severity))
+
+        entry = config_db.get_entry("SUPPRESS_ASIC_SDK_HEALTH_EVENT", severity)
+        need_remove = False
+
+        if category_list:
+            if suppressedCategoriesList:
+                entry["categories"] = suppressedCategoriesList
+            elif entry.get("categories"):
+                entry.pop("categories")
```

Comment:
no need pop?

Reply:
we do need it in the following scenario.
entry = {"categories": "cpu_hw", "max_entries": "10"}
cmd: `config asic-sdk-health-event suppress --categories none`
expected: {"max_entries": "10"}
in this scenario, `need_remove` is `true` but `entry` is not `None`. So code reaches line [7458](https://github.com/stephenxs/sonic-utilities/pull/44/files#diff-8bb4eed8e9e1f9974cbb7b46799411ecc318bae391f48735e7ad84a8122809a0R7458) `if entry:`. we need to keep `max_entries` but remove `categories`.